### ISSUE TITLE
indent doesn't apply to expressions split across multiple lines

### ISSUE THREAD

**Issue by Jamesernator** [eslint/eslint#12255]
**Tell us about your environment**

* **ESLint Version:** 6.3.0
* **Node Version:** 10.16.3
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// rules.js
"use strict";

module.exports = {
    // Error detection
    "no-undef": "error",
    "for-direction": "error",
    "getter-return": "error",
    "no-constant-condition": [
        "error",
        { checkLoops: false },
    ],
    "no-dupe-args": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-empty-character-class": "error",
    "no-ex-assign": "error",
    "no-extra-boolean-cast": "error",
    "no-func-assign": "error",
    "no-invalid-regexp": [
        "error",
        { allowConstructorFlags: ["s"] },
    ],
    "no-irregular-whitespace": "error",
    "no-obj-calls": "error",
    "no-prototype-builtins": "error",
    "no-regex-spaces": "error",
    "no-template-curly-in-string": "error",
    "no-unexpected-multiline": "error",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "use-isnan": "error",
    "valid-typeof": "error",
    "accessor-pairs": "error",
    "array-callback-return": "error",
    "block-scoped-var": "error",
    "consistent-return": "error",
    "no-compare-neg-zero": "error",
    eqeqeq: [
        "error",
        "always",
        { null: "ignore" },
    ],
    "no-implied-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-labels": "error",
    "no-extra-label": "error",
    "no-global-assign": "error",
    "no-lone-blocks": "error",
    "no-new-wrappers": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-redeclare": "error",
    "no-return-assign": "error",
    "no-self-assign": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-unmodified-loop-condition": "error",
    "no-unused-expressions": [
        "error",
        { allowTaggedTemplates: true },
    ],
    "no-unused-labels": "error",
    "no-useless-call": "error",
    "no-useless-concat": "error",
    "no-useless-escape": "error",
    "no-useless-return": "error",
    "no-useless-catch": "error",
    "no-with": "error",
    "prefer-promise-reject-errors": "error",
    radix: "error",
    strict: [
        "error",
        "global",
    ],
    "no-delete-var": "error",
    "no-shadow-restricted-names": "error",
    "no-unused-vars": [
        "error",
        {
            varsIgnorePattern: "^_",
            argsIgnorePattern: "^_",
        },
    ],
    "no-use-before-define": [
        "error",
        { classes: false, functions: false },
    ],
    "no-new-require": "error",
    "no-path-concat": "error",
    "unicode-bom": [
        "error",
        "never",
    ],
    "no-fallthrough": "error",
    "dot-notation": "error",
    "no-alert": "error",
    "no-async-promise-executor": "error",
    "no-misleading-character-class": "error",
    "require-atomic-updates": "error",
    "default-case": "error",
    "prefer-named-capture-group": "error",
    "require-unicode-regexp": "error",
    "no-buffer-constructor": "error",

    // ---- Style rules ----
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-extra-parens": [
        "error",
        "all",
        { nestedBinaryExpressions: false },
    ],
    "no-extra-semi": "error",
    "dot-location": [
        "error",
        "property",
    ],
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-implicit-globals": "error",
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-proto": "error",
    "no-void": "error",
    "no-label-var": "error",
    "array-bracket-spacing": "error",
    "brace-style": [
        "error",
        "1tbs",
        { allowSingleLine: true },
    ],
    camelcase: [
        "error",
        { properties: "never" },
    ],
    "comma-dangle": [
        "error",
        {
            arrays: "always-multiline",
            objects: "always-multiline",
            imports: "always-multiline",
            exports: "always-multiline",
            functions: "always-multiline",
        },
    ],
    "comma-spacing": [
        "error",
        { before: false, after: true },
    ],
    "comma-style": "error",
    "computed-property-spacing": "error",
    "eol-last": "error",
    "func-call-spacing": "error",
    "function-paren-newline": [
        "error",
        "consistent",
    ],
    indent: [
        "error",
        4,
        { ignoredNodes: ["ConditionalExpression"] },
    ],
    "key-spacing": [
        "error",
        { beforeColon: false, afterColon: true },
    ],
    "keyword-spacing": "error",
    "lines-around-comment": [
        "error",
        {
            allowBlockStart: true,
            allowArrayStart: true,
            allowObjectStart: true,
        // TODO: Check others
        },
    ],
    "max-len": [
        "error",
        120,
        {
            ignoreUrls: true,
            ignoreStrings: true,
            ignoreTemplateLiterals: true,
            ignoreRegExpLiterals: true,
            ignoreComments: true,
        },
    ],
    "max-statements-per-line": "error",
    "new-cap": [
        "error",
        { properties: false },
    ],
    "new-parens": "error",
    "newline-per-chained-call": [
        "error",
        { ignoreChainWithDepth: 3 },
    ],
    "no-array-constructor": "error",
    "no-bitwise": "error",
    "no-lonely-if": "error",
    "no-mixed-spaces-and-tabs": "error",
    "no-multi-assign": "error",
    "no-new-object": "error",
    "no-tabs": "error",
    "no-trailing-spaces": "error",
    "no-unneeded-ternary": "error",
    "no-whitespace-before-property": "error",
    "nonblock-statement-body-position": [
        "error",
        "beside",
    ],
    "object-curly-newline": [
        "error",
        { consistent: true },
    ],
    "object-curly-spacing": [
        "error",
        "always",
    ],
    "one-var": [
        "error",
        { initialized: "never" },
    ],
    "operator-assignment": "error",
    "padded-blocks": [
        "error",
        "never",
    ],
    "quote-props": [
        "error",
        "as-needed",
    ],
    "semi-spacing": "error",
    "semi-style": "error",
    "space-before-function-paren": [
        "error",
        {
            anonymous: "never",
            named: "never",
            asyncArrow: "always",
        },
    ],
    "space-unary-ops": [
        "error",
        { nonwords: false, words: true },
    ],
    "switch-colon-spacing": "error",
    "template-tag-spacing": [
        "error",
        "never",
    ],
    "arrow-parens": [
        "error",
        "always",
    ],
    "arrow-spacing": [
        "error",
        { before: true, after: true },
    ],
    "constructor-super": "error",
    "generator-star-spacing": [
        "error",
        "after",
    ],
    "no-class-assign": "error",
    "no-const-assign": "error",
    "no-duplicate-imports": "error",
    "no-new-symbol": "error",
    "no-this-before-super": "error",
    "no-useless-computed-key": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-var": "error",
    "object-shorthand": [
        "error",
        "always",
    ],
    "prefer-arrow-callback": [
        "error",
        {
            allowNamedFunctions: true,
            allowUnboundThis: true,
        },
    ],
    "prefer-const": [
        "error",
        { destructuring: "all" },
    ],
    "prefer-destructuring": [
        "error",
        {
            array: false,
            object: true,
        },
        {
            enforceForRenamedProperties: false,
        },
    ],
    "prefer-numeric-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "error",
    "symbol-description": "error",
    "template-curly-spacing": [
        "error",
        "always",
    ],
    "yield-star-spacing": [
        "error",
        { before: false, after: true },
    ],
    curly: [
        "error",
        "multi-line",
        "consistent",
    ],
    "line-comment-position": "error",
    "space-before-blocks": [
        "error",
        "always",
    ],
    "max-statements": [
        "error",
        500,
    ],
    complexity: [
        "error",
        { max: 40 },
    ],
    "no-inline-comments": "error",
    "block-spacing": "error",
    "spaced-comment": [
        "error",
        "always",
        { markers: ["/"] },
    ],
    "no-div-regex": "error",
    "no-new": "error",
    "linebreak-style": "error",
    "max-depth": "error",
    "max-nested-callbacks": "error",
    "max-params": [
        "error",
        4,
    ],
    "no-negated-condition": "error",
    "no-plusplus": "error",
    "no-empty": [
        "error",
        { allowEmptyCatch: false },
    ],
    "no-empty-function": "error",
    quotes: [
        "error",
        "double",
        { allowTemplateLiterals: true },
    ],
    "no-else-return": "error",
    "no-eval": "error",
    "no-new-func": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": "error",
    "no-throw-literal": "error",
    "array-element-newline": [
        "error",
        "consistent",
    ],
    "implicit-arrow-linebreak": [
        "error",
        "beside",
    ],
    "lines-between-class-members": [
        "error",
        "always",
        { exceptAfterSingleLine: true },
    ],
    "no-multiple-empty-lines": [
        "error",
        { max: 3, maxBOF: 1, maxEOF: 1 },
    ],
    "prefer-object-spread": "error",
    "rest-spread-spacing": [
        "error",
        "never",
    ],
    semi: [
        "error",
        "always",
    ],
    "function-call-argument-newline": [
        "error", 
        "consistent",
    ],

    // ---- Import errors ----

    "import/no-absolute-path": "error",
    "import/no-webpack-loader-syntax": "error",
    "import/no-self-import": "error",
    "import/no-useless-path-segments": "error",
    "import/no-unused-modules": "error",
    "import/no-mutable-exports": "error",
    "import/no-extraneous-dependencies": "error",
    "import/unambiguous": "error",
    "import/no-amd": "error",
    "import/named": "error",
    "import/default": "error",
    "import/namespace": [
        "error",
        {
            allowComputed: true,
        },
    ],
    "import/export": "error",
    "import/no-unresolved": "error",

    // TODO: Renable these ruels when they don't destroy performance
    "import/no-named-as-default": "off",
    "import/no-deprecated": "off",
    "import/no-cycle": "off",

    // ---- Import styles ----
    "import/no-dynamic-require": "error",
    "import/first": "error",
    "import/no-duplicates": "error",
    "import/extensions": [
        "error",
        "always",
        { ignorePackages: true },
    ],
    "import/order": [
        "error",
        { "newlines-between": "never" },
    ],
    "import/newline-after-import": "error",
    "import/prefer-default-export": "error",
    "import/no-named-default": "error",
    "import/no-anonymous-default-export": [
        "error",
        {
            allowObject: true,
            allowArray: false,
            allowArrowFunction: false,
            allowAnonymousClass: false,
            allowAnonymousFunction: false,
            allowCallExpression: false,
            allowLiteral: false,
        },
    ],

    // ---- Import off ----
    "import/no-relative-parent-imports": "off",
    "import/no-restricted-paths": "off",
    "import/no-internal-modules": "off",
    "import/no-commonjs": "off",
    "import/no-nodejs-modules": "off",
    "import/exports-last": "off",
    "import/no-namespace": "off",
    "import/max-dependencies": "off",
    "import/no-unassigned-import": "off",
    "import/no-default-export": "off",
    "import/no-named-export": "off",
    "import/group-exports": "off",
    "import/dynamic-import-chunkname": "off",

    // Disabled
    "no-nested-ternary": "off",
    "no-await-in-loop": "off",
    "valid-jsdoc": "off",
    "class-methods-use-this": "off",
    "no-inner-declarations": "off",
    "guard-for-in": "off",
    "no-eq-null": "off",
    "no-empty-pattern": "off",
    "no-invalid-this": "off",
    "no-iterator": "off",
    "no-loop-func": "off",
    "no-restricted-properties": "off",
    "no-script-url": "off",
    "require-await": "off",
    "init-declarations": "off",
    "no-catch-shadow": "off",
    "no-restricted-globals": "off",
    "no-undef-init": "off",
    "callback-return": "off",
    "global-require": "off",
    "handle-callback-err": "off",
    "no-mixed-require": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-restricted-modules": "off",
    "no-sync": "off",
    "capitalized-comments": "off",
    "consistent-this": "off",
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": "off",
    "id-blacklist": "off",
    "id-match": "off",
    "jsx-quotes": "off",
    "multiline-ternary": "off",
    "no-continue": "off",
    // TODO: Consider groups
    "no-mixed-operators": "off",
    "no-ternary": "off",
    "no-underscore-dangle": "off",
    "object-property-newline": "off",
    "one-var-declaration-per-line": "off",
    "require-jsdoc": "off",
    "sort-keys": "off",
    "wrap-regex": "off",
    "arrow-body-style": "off",
    "no-confusing-arrow": "off",
    "no-restricted-imports": "off",
    "require-yield": "off",
    "wrap-iife": "off",
    "no-sparse-arrays": "off",
    yoda: "off",
    "no-magic-numbers": "off",
    "no-shadow": "off",
    "no-return-await": "off",
    "id-length": "off",
    "no-warning-comments": "off",
    "array-bracket-newline": "off",
    "vars-on-top": "off",
    // TODO: Consider
    "padding-line-between-statements": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    // TODO: Implement
    "operator-linebreak": "off",
    console: "off",
    "max-classes-per-file": "off",
    "no-param-reassign": "off",
    "no-undefined": "off",
    // TODO: Consider
    "max-lines-per-function": "off",
    "no-restricted-syntax": "off",
    "sort-vars": "off",
    "max-lines": "off",
    "multiline-comment-style": "off",
    // Using eslint-plugin-import instead
    "sort-imports": "off",
    "no-dupe-class-members": "off",
};
```

```js
// config.js
'use strict';

module.exports = {
    root: true,
    plugins: [
        "import",
    ],
    env: {
        es6: true,
    },
    rules: require("./rules.cjs"),
};
```

</details>

**What is wrong?**

Eslint doesn't consider this to be an error:

```js
// example.js
const x
    = 1
+ 2;
```

**What did you expect to happen?**

I expected that `indent` would report that the line `+ 2;` should be indented.

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Idea solution**:

My idea for solving this is to add a new option to indent `multilineExpressions` that accepts an indent level similar to other such options:

<details>
<summary>e.g.</summary>

```js
// { multilineExpressions: 0 }
const x
  = 1
+ 2; // ERROR
```

```js
// { multilineExpressions: 0 }
const x
  = 1
  + 2; // OK
```

```js
// { multilineExpressions: 0 }
const x
  = 1
    + 2; // ERROR
```

```js
// { multilineExpressions: 1 }
const x
  = 1
+ 2; // ERROR
```

```js
// { multilineExpressions: 1 }
const x
  = 1
  + 2; // ERROR
```

```js
// { multilineExpressions: 1 }
const x
  = 1
    + 2; // OK
```

</details>

**Comment by kaicataldo** [comment]
This feels like a reasonable enhancement to see, but let's see what the rest of the team has to say.

Similar issue: https://github.com/eslint/eslint/issues/12248
